### PR TITLE
Feature/issue 56 manage tweet

### PR DESCRIPTION
**Pull request by g8rswimmer**
This a PR for issue #56 

https://developer.twitter.com/en/docs/twitter-api/tweets/manage-tweets/introduction

### CODE REVIEW

File: v2/client.go [hunk 1]
```diff
@@ -25,6 +25,96 @@ type Client struct {
 	Host       string
 }
 
+// CreateTweet will lets a user post polls, quote tweets, tweet with reply setting, tweet with geo, attach
+// perviously uploaded media toa tweet and tag users, tweet to super followers, etc.
```
Comment (Guitarbum722) [inline v2/client.go:29]: spelling and grammar if you care

File: v2/tweet_manage.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+// CreateTweetReply sets the reply setting for the tweet
+type CreateTweetReply struct {
+	ExcludeReplyUserIDs []string `json:"exclude_reply_user_ids"`
+	InReplyToTweetID    string   `json:"in_reply_to_tweet_id"`
+}
+
+func (r CreateTweetReply) validate() error {
+	if len(r.ExcludeReplyUserIDs) > 0 && len(r.InReplyToTweetID) == 0 {
+		return fmt.Errorf("reply in reply to tweet is needs to be present it exclude reply user ids are present %w", ErrParameter)
```
Comment (Guitarbum722) [inline v2/tweet_manage.go:74]: grammar (nitpick)

File: v2/tweet_manage.go [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+// CreateTweetMedia allows for updated media to attached.
+// If the tagged user ids are present, then ids must be present.
+type CreateTweetMedia struct {
+	IDs           []string `json:"media_ids"`
+	TaggedUserIDs []string `json:"tagged_user_ids"`
+}
+
+func (m CreateTweetMedia) validate() error {
+	if len(m.TaggedUserIDs) > 0 && len(m.IDs) == 0 {
+		return fmt.Errorf("media ids are required if taged user ids are present %w", ErrParameter)
```
Comment (Guitarbum722) [inline v2/tweet_manage.go:48]: ~taged~ tagged (nitpick)

**Review by Guitarbum722 — APPROVED** [review]
Nothing blocking...

**Comment by gdsoumya** [comment]
@g8rswimmer I was trying out the example for creating a tweet from the master branch but it appears that the authentication mech used (bearer tokens) won't work with that endpoint I get this error 
```
status 403 Unsupported Authentication:Authenticating with OAuth 2.0 Application-Only is forbidden for this endpoint.  Supported authentication types are [OAuth 1.0a User Context, OAuth 2.0 User Context]
```
Is it possible to provide an updated example or point out what I might be doing wrong?

**Comment by gdsoumya** [comment]
I was able to fix the auth issue by using an oauth1 pkg following this issue #15 but after that, the tweet API returned a `400 Invalid Request:One or more parameters to your request was invalid.`  I manually tinkered around and saw that the payload being sent in the body had some redundant/optional fields that were not being removed. I have raised a PR for the fix #84 , let me know if any changes are needed.

**Comment by g8rswimmer** [comment]
@gdsoumya thanks for the bug fix.  It would appear that auth 2.0 would need to use PKCE.  That might be an improvement to the examples in the future.

**Pull request merged**